core/src/test/java/org/testcontainers/containers/DockerComposeFileValidationTest.java

```diff
@@ -0,0 +1,70 @@
+package org.testcontainers.containers;
+
+import com.google.common.collect.ImmutableMap;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import java.io.File;
+
+import static java.util.Collections.emptyList;
+
+public class DockerComposeFileValidationTest {
+
+    @Test
+    public void shouldValidate() {
+        File file = new File("src/test/resources/docker-compose-container-name-v1.yml");
+        try (DockerComposeContainer container = new DockerComposeContainer<>(file)) {
+            Assertions.assertThatThrownBy(container::start)
+                .hasMessageContaining(file.getAbsolutePath())
+                .hasMessageContaining("'container_name' property set for service 'redis'");
+        }
+    }
+
+    @Test
+    public void shouldRejectContainerNameV1() {
+        Assertions
+            .assertThatThrownBy(() -> {
+                DockerComposeContainer.validate(ImmutableMap.of(
+                    "redis", ImmutableMap.of("container_name", "redis")
+                ), "");
+            })
+            .hasMessageContaining("'container_name' property set for service 'redis'");
+    }
+
+    @Test
+    public void shouldRejectContainerNameV2() {
+        Assertions
+            .assertThatThrownBy(() -> {
+                DockerComposeContainer.validate(ImmutableMap.of(
+                    "version", "2",
+                    "services", ImmutableMap.of(
+                        "redis", ImmutableMap.of("container_name", "redis")
+                    )
+                ), "");
+            })
+            .hasMessageContaining("'container_name' property set for service 'redis'");
+    }
+
+    @Test
+    public void shouldIgnoreUnknownStructure() {
```

Comment:
Why?
For possible forward compatibility?

Reply:
Yes. We should rather do "best effort" detection rather then making it fail on parsing